lib/commands/misc/giverole.js

```diff
@@ -26,14 +26,38 @@ module.exports = class GiveRoleCommand extends Command {
 
 	async run(msg, args) {
     if (args.role === "member") {
-      const role = msg.guild.roles.find(r => r.name == 'Member')
-      if (!msg.member.roles.has(role)) {
+      const role = msg.guild.roles.find(r => r.name == 'Member');
+
+      if (!msg.member.roles.has(role.id)) {		// Needs to use role.id to properly check if the user has that role -Syn
         msg.member.addRole(role);
         const channel = this.client.channels.get(getChannel(GENERAL_ID));
         channel.send(`Welcome to the server, ${msg.author}! <:wutGiga:297897855727697921>`);
       }
-    } else {
-      msg.reply(`Invalid role! Available options are: member`)
+
+			// Prevents this generalized code from allowing random people to self-assign the admin, moderator, or bae roles.
+    } else if (COLORS.includes(args.role)) {
+
+				const channel = this.client.channels.get(getChannel(REOL_CHAN_ID));
+				const role = msg.guild.roles.find(r => r.name == args.role);
+				const memberRole = msg.guild.roles.find(r => r.name == 'Member');
```

Comment:
It might be best to set up a helper of some sorts that gets these 3 values as we're using it a number of times in this PR (this'll reduce code duplication)